### ISSUE TITLE
[Bug]: Constant script crashing causing major studdering a lagging from the script VM.

### ISSUE THREAD

**Issue by gir489returns** [YimMenu/YimMenu#2462]
### Describe the bug

Something is wrong with the Script VM that is causing constant exceptions from the script runner. Seemed to happen just after #2456 @maybegreat48 Can you please look into this?

### Steps To Reproduce

Unknown replication steps, seems to happen at random, also happens in SP.

### Expected Behavior

Shouldn't slow the game down with the constant exceptions.

### Logs and/or screenshots

[Logs.zip](https://github.com/YimMenu/YimMenu/files/13502277/Logs.zip)

### Store

Rockstar

### Language

English

### YimMenu Language

English (US)

### Build

- [X] I've confirmed the issue exists on the latest version of YimMenu

### Coload

- [X] I've tested this without co-loading anything.

**Comment by lonelybud** [comment]
i think you cant even join the heist alone as a job though mobile also.

**Comment by gir489returns** [comment]
> i think you cant even join the heist alone as a job though mobile also.

Are you getting this same behavior? Can you please try #2461 to see if I fixed it?

**Comment by lonelybud** [comment]
> > i think you cant even join the heist alone as a job though mobile also.
> 
> Are you getting this same behavior? Can you please try #2461 to see if I fixed it?

i am not experiencing the bug as described in ticket. I was just asking that are you able to join heist via phone -

> go to the Quick Job menu on the phone. Select “Play Heist”

**Comment by gir489returns** [comment]
> > > i think you cant even join the heist alone as a job though mobile also.
> > 
> > 
> > Are you getting this same behavior? Can you please try #2461 to see if I fixed it?
> 
> i am not experiencing the bug as described in ticket. I was just asking that are you able to join heist via phone -
> 
> > go to the Quick Job menu on the phone. Select “Play Heist”

I can get into the lobby, however, I am unable to actually play, it gets stuck at launching session.

![image](https://github.com/YimMenu/YimMenu/assets/100792176/f61bce4c-fccb-4e98-a860-8fd4a5eb56a5)

**Comment by lonelybud** [comment]
> > > > i think you cant even join the heist alone as a job though mobile also.
> > > 
> > > 
> > > Are you getting this same behavior? Can you please try #2461 to see if I fixed it?
> > 
> > 
> > i am not experiencing the bug as described in ticket. I was just asking that are you able to join heist via phone -
> > > go to the Quick Job menu on the phone. Select “Play Heist”
> 
> I can get into the lobby, however, I am unable to actually play, it gets stuck at launching session.

this is the problem i faced. In my fork when i removed the native_detour `START_NEW_SCRIPT_WITH_ARGS`  for both am_launcher & freemode, i was able to join thereafter. I dont know which one is troublesome, also i think this is some another bug.

**Comment by gir489returns** [comment]
> this is the problem i faced. In my fork when i removed the native_detour `START_NEW_SCRIPT_WITH_ARGS` for both am_launcher & freemode, i was able to join thereafter. I dont know which one is troublesome, also i think this is some another bug.

I believe there was a much greater issue happening here with the metrics and the anti-cheat runners or possibly something else. Sometimes I am able to make it into the heist, but when I get in-game everything is frozen until I uninject and reinject, so that tells me a hook or something is rejecting something it's not supposed to.

**Comment by gir489returns** [comment]
I reverted head to https://github.com/YimMenu/YimMenu/commit/a5fb18c06b7c6991965eafc984a2378411ca413d and I was able to play heists just fine, I think there's something wrong with what we did after that, it's either metrics hook or the code to fix nightclub weapons.

**Comment by gir489returns** [comment]
> nightclub and jittering not an issue. What about this commit does it look fine [5c3b50c](https://github.com/YimMenu/YimMenu/commit/5c3b50cab7c43c4001ddd97bbe906eb3048159b5) ?

I already told you the hook is not called.

**Comment by lonelybud** [comment]
you are right

**Comment by lonelybud** [comment]
shall there be a separate ticket for this?

**Comment by lonelybud** [comment]
may be its a bug at gta end. i was able to join cayo easily. diamond casino failed.

**Comment by gir489returns** [comment]
No, something we're doing is contaminating the script stack.
